File: interview1.java
package leetcode;

public class interview1 {

    public static int search(int arr[]) {
        int low = 0;
        int high = arr.length - 1;
        int mid=0;

        while (low < high) {
            mid = low + (high - low) / 2;
            if ((arr[high]-arr[low])==2) {
                return arr[low] + 1;
            } else if ((arr[mid] - arr[low]) > (arr[high] - arr[mid])) {
                high = mid ;
            } else {
                low = mid ;
            }
        }
        return arr[low] + 1;
    }

    public static void main(String[] args) {
        int[] arr = {1, 2,4, 5, 6,7,8};
        int miss = search(arr);
        System.out.println(miss);
    }
}

File: interview1.cpp
/*array of n-1 integers and 1 value is missing return which one.
 example [1,2,4,5,6,7]
 output-3
 
 */
// Time Complexity : O(logN)
// Space Complexity :O(1)
// Did this code successfully run on Leetcode : yes
// Any problem you faced while coding this : no

#include <iostream>
#include<vector>

using namespace std;

int search(vector<int>& nums){
    if(nums.empty() || nums.size() == 0){
        return -1;
    }
    int low = 0;
    int high = nums.size() - 1;
    while(low<=high){
       int mid = low +(high - low) / 2;
       if(nums[mid-1] != (nums[mid] - 1)){
            return (nums[mid] - 1);
       }
       else if(nums[mid+1] != nums[mid]+1){
           return nums[mid]+1;
       }
       else if(nums[mid] > mid+1){
           high = mid-1;
       }
       else{
           low = mid+1;
       }
    }
    return -1;
    
}

int main()
{
    vector<int> val = {1,2,3,4,5,7};
    int x = search(val);
    std::cout << x << std::endl;
    return 0;
}
